### ISSUE TITLE
`.via_dt` could handle `NaT` better

### ISSUE THREAD

**Issue by gettodaze** [static-frame/static-frame#765]
**Description**
I am not sure the best way to handle NaT in `.via_dt` applications, but I think the current approach is not ideal. Ideally, I would think an `NaT.year` or `NaT.month` would return `NaN`.

**Example**
```python
In [1]: s = sf.Series(['2020-02', None, None], dtype='datetime64[M]')           

In [2]: s                                                                       
Out[2]: 
<Series>
<Index>
0        2020-02
1        NaT
2        NaT
<int64>  <datetime64[M]>

In [3]: s.via_dt.year                                                           
Out[3]: 
<Series>
<Index>
0        2020
1        -9223372036854773838
2        -9223372036854773838
<int64>  <int64>

In [4]: s.via_dt.month                                                          
Out[4]: 
<Series>
<Index>
0        2
1        5
2        5
<int64>  <int64>
```

**Platform**

Run the following function (static-frame >= 0.8.1) and provide the results to define your platform and environment:

```python
>>> import static_frame as sf
>>> sf.Platform.display()
<Series: platform>
<Index>
platform           Linux-5.15.0-67-generic-x86_64-with-glibc2.29
sys.version        3.8.12 (default, Mar 10 2023, 09:06:31) [GCC 9.4.0]
static-frame       0.9.16
numpy              1.23.4
pandas             1.5.2
xlsxwriter         <ModuleNotFoundError>
openpyxl           <ModuleNotFoundError>
xarray             <ModuleNotFoundError>
tables             <ModuleNotFoundError>
pyarrow            <ModuleNotFoundError>
msgpack            <ModuleNotFoundError>
msgpack_numpy      <ModuleNotFoundError>
<<U13>             <object>

```

**Comment by flexatone** [comment]
Many thanks, @gettodaze , for posting this issue. What you show is clearly a bad result! We could return NaN for NaT, but that would change the type to float which might be surprising. Another option is to expose the opportunity of supplying a `fill_value` somewhere... but that is tricky as these are all properties. Any other ideas on this might be handled?

**Comment by flexatone** [comment]
@gettodaze : The solution I am implementing offers an alternative `via_dt` interface that is callable, such that a `fill_value` argument can be provided. This preserves the usage of attributes on `via_dt`. See the example below; feedback is welcome.

```python
>>> s = sf.Series(['2020-02', None, None], dtype='datetime64[M]')
>>> s.via_dt(fill_value=-1).year
<Series>
<Index>
0        2020
1        -1
2        -1
<int64>  <int64>
```

**Comment by gettodaze** [comment]
I think that could work. And would `fill_value` default to `np.nan`?

**Comment by gettodaze** [comment]
I know some calls error for invalid values (like below), could that be an option?
```
In [13]: s.via_dt.fromisoformat()                                                                        
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-13-2f3fb7d17a20> in <module>
----> 1 s.via_dt.fromisoformat()

~/.env38/lib/python3.8/site-packages/static_frame/core/node_dt.py in fromisoformat(self)
    576                 yield array
    577 
--> 578         return self._blocks_to_container(blocks())
    579 
    580     def strftime(self, format: str) -> TContainer:

~/.env38/lib/python3.8/site-packages/static_frame/core/series.py in blocks_to_container(blocks)
    780         def blocks_to_container(blocks: tp.Iterator[np.ndarray]) -> 'Series':
    781             return self.__class__(
--> 782                 next(blocks), # assume only one
    783                 index=self._index,
    784                 name=self._name,

~/.env38/lib/python3.8/site-packages/static_frame/core/node_dt.py in blocks()
    565             for block in self._blocks:
    566                 # permit only string types, or objects types that contain strings
--> 567                 self._validate_dtype_str(block.dtype)
    568 
    569                 # NOTE: might use fromisoformat on date/datetime objects directly; assumed to be faster to go through datetime64 objects, fromisoformat is only available on python 3.7

~/.env38/lib/python3.8/site-packages/static_frame/core/node_dt.py in _validate_dtype_str(dtype, exclude)
    136                 ):
    137             return
--> 138         raise RuntimeError(f'invalid dtype ({dtype}) for operation on string types')
    139 
    140     #---------------------------------------------------------------------------

RuntimeError: invalid dtype (datetime64[M]) for operation on string types
```

**Comment by flexatone** [comment]
I think you are thinking of cases more like this: here the `fill_value` will be returned before a method is called, avoiding the error your were expecting:

```python
>>> s.astype('datetime64[D]').via_dt(fill_value=-1).isoformat()
<Series>
<Index>
0        2020-02-01
1        -1
2        -1
<int64>  <object>
```
